draft-barnes-mls-protocol.md

```diff
@@ -541,11 +554,25 @@ DH(DH(AB), DH(CD))
 A    B    C    D
 ~~~~~
 
-Ratchet trees constructed this way provide the property that one must hold at
-least one private key from the tree to compute the secret root key. With all
-participants holding one leaf private key; this allows any individual to update
-their own key and change the shared root key, such that only group members can
-compute the new key.
+### Ratchet Trees for TreeKEM
+
+The key computations for each Node differs in TreeKEM. The general idea behind
+TreeKEM is to obtain better messaging complexity on concurrent events by not
+forcing contributivity of all leaves to the keys of the *intermediate* nodes.
+Instead, modification of a key will lead to sending to the copath the
+information necessary to be able to recompute the root key through a
+Key Encapsulation Mechanism (KEM) which can be typically post-quantum resistant.
+
+* parent_secret = Hash(A OR B)
+* parent_private, parent_public = Derive-Key-Pair(parent_secret)
+
+~~~~~
+   H(H(A||B)||H(C||D))
+       /      \
+   H(A||B)   H(C||D)
+    /  \      /  \
+   A    B    C    D
```

Comment:
This is kind of confusing. I think you need to start small. Why don't you crib my text from the mail to the listl.